Engine/ac/string.cpp

```diff
@@ -49,6 +49,67 @@ int String_IsNullOrEmpty(const char *thisString)
     return 0;
 }
 
+void calculate_joint_length(const size_t count, const uint32_t *&handles, int sep_len, int sep_ulen, int &total_len, int &total_ulen)
```

Comment:
I notice that you iterate over handles and do `ccGetObjectAddressFromHandle` on them twice in succession, first to calculate final length and second to actually do string copying. This seem suboptimal and complicates the code. You could instead make a function that gathers the resolved ScriptString's pointers into a std::vector, and then do whatever you want with them: get their headers, check lengths, then copy their contents.

Reply:
I got rid of the function but wasn't sure on making it a function because then I would iterate three times on the strings, because I still need to figure the length before I can create a buffer...

Reply:
Hmm... well, the function itself would iterate over handles and call ccGetObjectAddressFromHandle in order to gather strings, that may come handy later. Other 2 loops would be over a ready vector of string pointers, I don't think that's too bad.